### PR TITLE
Fix #10656 return value for HttpServletRequest.getProtocolRequestId() in ee10

### LINKED ISSUE #10656: EE10 `ServletRequest.getProtocolRequestId()` impl not spec compliant  when protocol is H1
**Jetty version(s)**
12.0.x

**Jetty Environment**
EE10

**Description**
The `jakarta.servlet.ServletRequest.getProtocolRequestId()` javadoc states that when the protocol is HTTP 1.x, the empty string should be returned.

The implementation in `ServletApiRequest.getProtocolRequestId()` delegates the call to `ChannelRequest.getId()`, which got the ID from `HttpStreamOverHTTP1.getId()`, the latter always having some non-empty string.

### PR DESCRIPTION
**Pull request by gregw**
Return null for HTTP/1.0 and previous

Fixes: #10656

### CODE REVIEW

File: jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/ServletApiRequest.java [hunk 1]
```diff
@@ -187,6 +187,8 @@ public String getRequestId()
     @Override
     public String getProtocolRequestId()
     {
+        if (getRequest().getConnectionMetaData().getHttpVersion().getVersion() <= HttpVersion.HTTP_1_0.getVersion())
```
Comment (lorban) [inline jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/ServletApiRequest.java]: This should be `<= HttpVersion.HTTP_2_0.getVersion()` as specified in this method's javadoc.

File: jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/ServletApiRequest.java [hunk 2]
```diff
@@ -187,6 +187,8 @@ public String getRequestId()
...
+            return null;
```
Comment (lorban) [inline jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/ServletApiRequest.java]: This should be `return "";` as specified in this method's javadoc.

**Review by lorban — CHANGES_REQUESTED** [review]

File: jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/ServletApiRequest.java (hunk 1 above)
Comment (gregw) [reply jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/ServletApiRequest.java]: I've switched to only return protocol IDs for protocols known to have IDs

**Review by lorban — APPROVED** [review]

**Comment by gregw** [comment]
The CI failures are unrelated and I see them also in other unrelated PRs, so I'm going to use my superpowers and merge anyway

**Pull request merged**